fabtests/component/dmabuf-rdma/fi-rdmabw-xe.c

```diff
@@ -548,6 +551,65 @@ static void finalize_ofi(void)
 	}
 }
 
+/*
+ * Context pool to support FI_CONTEXT mode
+ */
+
+static struct context_list {
+	struct fi_context context;
+	struct context_list *next;
+} free_list[TX_DEPTH + 1], *free_head, *free_tail;
+
+static bool context_pool_initialized = false;
+
+static void init_context_pool(void)
+{
+	int i;
+
+	if (context_pool_initialized)
+		return;
+
+	free_head = &free_list[0];
+	free_tail = &free_list[TX_DEPTH];
+	for (i = 0; i < TX_DEPTH; i++)
+		free_list[i].next = &free_list[i+1];
+
+	context_pool_initialized = true;
+}
+
+static struct fi_context *get_context(void)
+{
+	struct context_list *entry;
+
+	init_context_pool();
+
+	if (free_head == free_tail) {
+		printf("%s: can't alloc fi_context!\n", __func__);
+		return NULL;
+	}
+
+	entry = free_head;
+	free_head = free_head->next;
+
+	entry->next = NULL;
+	return &entry->context;
+}
+
+static void put_context(struct fi_context *ctxt)
+{
+	struct context_list *entry;
+
+	assert(context_pool_initialized);
+
+	if (!ctxt)
+		return;
+
+	entry = container_of(ctxt, struct context_list, context);
+	entry->next = NULL;
+	free_tail->next = entry;
+	free_tail = entry;
+}
```

Comment:
Can we move this to a common area?

Reply:
Like where? The component tests are mostly self-contained. Separating this part into its own file is fine, but I am reluctant to put this into `common/share.c`.

Reply:
Thanks. Will do.